projects/revest/index.js

```diff
@@ -1,81 +1,211 @@
-    chainId: 137,
-  },
-  fantom: {
-    owner: ['0x3923E7EdBcb3D0cE78087ac58273E732ffFb82cf',],
-    chainId: 250,
-  },
-  avax: {
-    owner: ['0x955a88c27709a1EEf4ACa0df0712c67B48240919',],
-    chainId: 43114,
-  },
+const { ethers } = require("ethers");
+
+const HOLDERS = {
+  ethereum: "0xA81bd16Aa6F6B25e66965A2f842e9C806c0AA11F",
+  polygon: "0x3cCc20d960e185E863885913596b54ea666b2fe7",
+  fantom: "0x3923E7EdBcb3D0cE78087ac58273E732ffFb82cf",
+  avax: "0x955a88c27709a1EEf4ACa0df0712c67B48240919",
+};
+
+const MIN_BLOCK = {
+  ethereum: 0,
+  polygon: 0,
+  fantom: 32894036,
+  avax: 0,
+};
+
+const eventLog_addresses = {
+  fantom: ["0xb80f5a586BC247D993E6dbaCD8ADD211ec6b0cA5"],
+};
+
+const tokenAddresses = {
+  fantom: { LQDR: "0x10b620b2dbAC4Faa7D7FFD71Da486f5D44cd86f9" },
+};
+
+const providers = {
+  fantom: "https://rpc.ftm.tools/",
+};
+
+async function mainnetTVL(time, block) {
+  const tokenRes = await axios.get(
+    "https://defi-llama-feed.vercel.app/api/address"
+  );
+  const balances = {};
+  await calculateTVL(tokenRes, balances, block);
+  return balances;
 }
 
-const getTokens = async chainId => {
-  let url = 'https://defi-llama-feed.vercel.app/api/address'
-  if (chainId) url = `${url}?chainId=${chainId}`
-  return (await axios.get(url)).data.body
+async function polygonTVL(time, block) {
+  const tokenRes = await axios.get(
+    "https://defi-llama-feed.vercel.app/api/address?chainId=137"
+  );
+  const balances = {};
+  const transform = await transformPolygonAddress();
+  await calculateTVL(tokenRes, balances, block, "polygon", transform);
+  return balances;
 }
 
-const moduleExports = Object.keys(chainConfig).reduce((agg, chain) => {
-  const { chainId, owner, pool2Token, rvst } = chainConfig[chain]
+async function fantomTVL(time, block) {
+  const tokenRes = await axios.get(
+    "https://defi-llama-feed.vercel.app/api/address?chainId=250"
+  );
+
+  const balances = {};
+  const transform = await transformFantomAddress();
+  await calculateTVL(tokenRes, balances, block, "fantom", transform);
 
-  let staking, pool2
-  async function tvl(ts, _block, chainBlocks) {
-    const block = chainBlocks[chain]
-    const transformAddress = await getChainTransform(chain)
-    const tokens = (await getTokens(chainId)).map(token => token.toLowerCase()).filter(token => ![rvst, pool2Token].includes(token))
+  for (var x = 0; x < eventLog_addresses["fantom"].length; x++) {
+    let event_balance = await parse_event_logs_for_balance(
+      "fantom",
+      eventLog_addresses["fantom"][x]
+    );
 
-    const balances = {}
-    const tokensAndOwners = []
-    tokens.map(token => owner.map(o => tokensAndOwners.push([token, o])))
-    await sumTokens(balances, tokensAndOwners, block, chain, transformAddress, { resolveCrv: true, resolveLP: true, resolveYearn: true })
-    return balances
+    //TBH i'm not sure this is the intended solution but i'm rolling with it anyways
```

Comment:
yeah, this part worries me a bit like as these logs would only over time and become heavy and break the adapter. I looked into the code but I dont see any other way to compute, but also I am a noob, is there a less intensive way of doing this, maybe via a graph query somewhere?

(Thanks for fixing the merge conflict)

Reply:
you could probably use the graph to compute these things but it also makes the code but more unruly and will take longer. That's a Rob question.

Reply:
@g1nt0ki I have rolled back the remote-branch to the last commit before the merge so we can coordinate on resolving the conflicts going forward. My bad